![logo-gt.png](attachment:logo-gt.png)

<div class="alert alert-block alert-info">
    <h1>BIOS 4150/BIOL 6150</h1>
    <h3>Instructor: Dr. King Jordan</h3>
    <p>Shivam Sharma (shivamsharma13@gatech.edu), Nilavrah Sensarma (nsensarma3@gatech.edu), Bengy Gyimah-Asamoah, (bgyimaha3@gatech.edu)

</p>
</div>

<div class="alert alert-block alert-warning">
    <h2>Project 6 (Variant consequences) starter notebook</h2>
    <h3>Deadline: 11:59PM, November 5th, 2024</h3>
</div>

<div class="alert alert-block alert-danger">
    <h2>Alert!</h2>
    <p>Do not delete any comments or information boxes the starter notebook comes with</p>
</div>

---

## 1. Quantifying by populations
### *Total Questions: 2*
### *Total Points: 25+40 (10+30) = 65*



In [1]:
#ls -lh the 1000 genomes source files
!ls -lh /storage/ice-shared/biol6150/Data/1000Genomes/

total 15G
-rw-r--r--. 1 ssharma454 gtperson 742M Oct 25 10:20 phase3.chr10.GRCh38.GT.crossmap.vcf.gz
-rw-r--r--. 1 ssharma454 gtperson 123K Oct 26 16:30 phase3.chr10.GRCh38.GT.crossmap.vcf.gz.tbi
-rw-r--r--. 1 ssharma454 gtperson 736M Oct 25 10:20 phase3.chr11.GRCh38.GT.crossmap.vcf.gz
-rw-r--r--. 1 ssharma454 gtperson 123K Oct 25 10:30 phase3.chr11.GRCh38.GT.crossmap.vcf.gz.tbi
-rw-r--r--. 1 ssharma454 gtperson 711M Oct 25 10:20 phase3.chr12.GRCh38.GT.crossmap.vcf.gz
-rw-r--r--. 1 ssharma454 gtperson 122K Oct 26 16:42 phase3.chr12.GRCh38.GT.crossmap.vcf.gz.tbi
-rw-r--r--. 1 ssharma454 gtperson 534M Oct 25 10:20 phase3.chr13.GRCh38.GT.crossmap.vcf.gz
-rw-r--r--. 1 ssharma454 gtperson  90K Oct 26 16:41 phase3.chr13.GRCh38.GT.crossmap.vcf.gz.tbi
-rw-r--r--. 1 ssharma454 gtperson 482M Oct 25 10:20 phase3.chr14.GRCh38.GT.crossmap.vcf.gz
-rw-r--r--. 1 ssharma454 gtperson  84K Oct 26 16:41 phase3.chr14.GRCh38.GT.crossmap.vcf.gz.tbi
-rw-r--r--. 1 ssharma454 gtperson 439M Oct 25 10:20 phase3.c

<div class="alert alert-block alert-warning">
    <h3>1.1 Sickle cell carriers by population</h3>
    <p><b>Background: </b>Different genes perform different functions in our body. rs334 is a missense variant on HBB gene, involved in haemoglobin pathway. It is associated with sickle cell anaemia and malaria resistance, and represents the phenotype-associated A allele.</p>
    <p><b>Objective: </b>The 1000 genomes VCF files have multiple samples in them. Every sample has a designated population label such as GWD, MSL, PUR, LWK, CEU, ACB, and so on. Find out the number of people in every population who carry the sickle cell variant. The final output will simply show number of samples per poulation (you can write it in markdown text).</p>
    <p><b>Instructions: </b>We expect you to show all your work (VCF file processing; extraction etc). You can use Python programming for any data processing. </p>
        <p>There is no specific formatting requirement for the final answer. But it should be readable and comprehensible to the notebook reader.</p>
</div>

In [ ]:
# Put all your work for 1.1 here.

In [1]:
#make a directory for the variant files
!mkdir -p ~/biol6150/ProjectSubmissions/Group8/Project6/Question1

In [1]:
#download the VCF file from 1000 genomes 
#the sickle cell variant, HBB, is on chromosome 11, so download the chromosome 11 files
!wget https://ftp.1000genomes.ebi.ac.uk/vol1/ftp/release/20130502/ALL.chr11.phase3_shapeit2_mvncall_integrated_v5b.20130502.genotypes.vcf.gz

--2024-11-04 21:30:12--  https://ftp.1000genomes.ebi.ac.uk/vol1/ftp/release/20130502/ALL.chr11.phase3_shapeit2_mvncall_integrated_v5b.20130502.genotypes.vcf.gz
Resolving ftp.1000genomes.ebi.ac.uk (ftp.1000genomes.ebi.ac.uk)... 193.62.193.167
Connecting to ftp.1000genomes.ebi.ac.uk (ftp.1000genomes.ebi.ac.uk)|193.62.193.167|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 734583405 (701M) [application/x-gzip]
Saving to: ‘ALL.chr11.phase3_shapeit2_mvncall_integrated_v5b.20130502.genotypes.vcf.gz’

ALL.chr11.phase3_sh 100%[===================>] 700.55M  47.8MB/s    in 13s     

2024-11-04 21:30:27 (54.1 MB/s) - ‘ALL.chr11.phase3_shapeit2_mvncall_integrated_v5b.20130502.genotypes.vcf.gz’ saved [734583405/734583405]



In [1]:
#move the file
!mv ~/biol6150/ProjectSubmissions/Group8/Project6/ALL.chr11.phase3_shapeit2_mvncall_integrated_v5b.20130502.genotypes.vcf.gz ~/biol6150/ProjectSubmissions/Group8/Project6/Question1

In [1]:
#check the format of the file
!bcftools view -h  ~/biol6150/ProjectSubmissions/Group8/Project6/Question1/ALL.chr11.phase3_shapeit2_mvncall_integrated_v5b.20130502.genotypes.vcf.gz | grep "FORMAT=<ID=GT"

##FORMAT=<ID=GT,Number=1,Type=String,Description="Genotype">


In [1]:
#create a vcf file containing only those with the HBB variant
!bcftools view -i 'ID="rs334"' ~/biol6150/ProjectSubmissions/Group8/Project6/Question1/ALL.chr11.phase3_shapeit2_mvncall_integrated_v5b.20130502.genotypes.vcf.gz -o ~/biol6150/ProjectSubmissions/Group8/Project6/Question1/HBB_variant.vcf


In [1]:
#extract only the identifiers into a text file
!bcftools query -l ~/biol6150/ProjectSubmissions/Group8/Project6/Question1/HBB_variant.vcf > ~/biol6150/ProjectSubmissions/Group8/Project6/Question1/hbb_identifiers.txt

In [1]:
#download the population info file
!wget https://ftp.1000genomes.ebi.ac.uk/vol1/ftp/release/20130502/integrated_call_samples_v3.20130502.ALL.panel

--2024-11-04 22:16:45--  https://ftp.1000genomes.ebi.ac.uk/vol1/ftp/release/20130502/integrated_call_samples_v3.20130502.ALL.panel
Resolving ftp.1000genomes.ebi.ac.uk (ftp.1000genomes.ebi.ac.uk)... 193.62.193.167
Connecting to ftp.1000genomes.ebi.ac.uk (ftp.1000genomes.ebi.ac.uk)|193.62.193.167|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 55156 (54K)
Saving to: ‘integrated_call_samples_v3.20130502.ALL.panel’

integrated_call_sam 100%[===================>]  53.86K   191KB/s    in 0.3s    

2024-11-04 22:16:46 (191 KB/s) - ‘integrated_call_samples_v3.20130502.ALL.panel’ saved [55156/55156]



In [1]:
#move it to the folder
!mv ~/biol6150/ProjectSubmissions/Group8/Project6/integrated_call_samples_v3.20130502.ALL.panel ~/biol6150/ProjectSubmissions/Group8/Project6/Question1/integrated_call_samples_v3.20130502.ALL.panel

In [1]:
#extracting the identifier and population data from the panel file
import pandas as pd
pop_info = pd.read_csv('~/biol6150/ProjectSubmissions/Group8/Project6/Question1/integrated_call_samples_v3.20130502.ALL.panel', sep='\t')
print(pop_info.columns)
sample_pop_data = pop_info[['sample', 'pop']]
#create a 2-column csv with sample and pop
sample_pop_data.to_csv('sample_pop_data.csv', index=False)
print("done")

#cross reference the identifiers to the sample populations
#load population
population_mapping = pd.read_csv('sample_pop_data.csv')
# Load identifiers
identifiers = pd.read_csv('~/biol6150/ProjectSubmissions/Group8/Project6/Question1/hbb_identifiers.txt', header=None, names=['identifier'])
print("done")

#merge and save as a csv
merged_data = pd.merge(identifiers, population_mapping, left_on='identifier', right_on='sample', how='left')
merged_data.to_csv('merged_data.csv', index=False)
print("done")

#count the number in each population
population_counts = merged_data.groupby('pop').size().reset_index(name='counts')
print(population_counts)
print("done")

Index(['sample', 'pop', 'super_pop', 'gender', 'Unnamed: 4', 'Unnamed: 5'], dtype='object')
done
done
done
    pop  counts
0   ACB      96
1   ASW      61
2   BEB      86
3   CDX      93
4   CEU      99
5   CHB     103
6   CHS     105
7   CLM      94
8   ESN      99
9   FIN      99
10  GBR      91
11  GIH     103
12  GWD     113
13  IBS     107
14  ITU     102
15  JPT     104
16  KHV      99
17  LWK      99
18  MSL      85
19  MXL      64
20  PEL      85
21  PJL      96
22  PUR     104
23  STU     102
24  TSI     107
25  YRI     108
done


In [1]:
#cleaning things up a bit
#unused files
!mv ~/biol6150/ProjectSubmissions/Group8/Project6/ALL.chr11.phase3_shapeit2_mvncall_integrated_v5b.20130502.genotypes.vcf.gz.1 !mv ~/biol6150/ProjectSubmissions/Group8/Project6/Question1/IGNORE/
!mv ~/biol6150/ProjectSubmissions/Group8/Project6/sickle_cell_genotypes_true.txt !mv ~/biol6150/ProjectSubmissions/Group8/Project6/Question1/IGNORE/
!mv ~/biol6150/ProjectSubmissions/Group8/Project6/sickle_cell_genotypes.txt !mv ~/biol6150/ProjectSubmissions/Group8/Project6/Question1/IGNORE/

#used files
!mv ~/biol6150/ProjectSubmissions/Group8/Project6/merged_data.csv !mv ~/biol6150/ProjectSubmissions/Group8/Project6/Question1/
!mv ~/biol6150/ProjectSubmissions/Group8/Project6/sample_pop_data.csv !mv ~/biol6150/ProjectSubmissions/Group8/Project6/Question1/

mv: cannot stat '!mv': No such file or directory
mv: cannot stat '!mv': No such file or directory
mv: cannot stat '!mv': No such file or directory
mv: cannot stat '!mv': No such file or directory
mv: cannot stat '!mv': No such file or directory


<div class="alert alert-block alert-warning">
    <h3>1.2 CFTR and cystic fibrosis</h3>
    <p><b>Background: </b>The cystic fibrosis transmembrane conductance regulator (CFTR) protein helps to maintain the balance of salt and water on many surfaces in the body, such as the surface of the lung. The CFTR protein is a particular type of protein called an ion channel.</p><br>
    <p><b>Objective 1.2.1: </b>How many Pathogenic variants are present in the 1000 Genomes data?</p>
    <p><b>Objective 1.2.2: </b>Give a population breakdown of individuals who have at least 1 Pathogenic CFTR variants as defined in the Gnomad browser? Your final answer can look like: CEU: 5, TSI: 2, ...</p><br>
    <p><b>Instructions: </b>"Pathogenic" can be defined in a variety of ways. For simplicity, we will use the Gnomad browser in this case. You can search any gene in the browser, and a detailed table is provided at the end of the search page (check the screenshot below). Clinical significance based on ClinVar annotations are provided in this table.</p>
    <p>Please show all your work. There is no specific formatting requirement for the final answer. But it should be readable and comprehensible to the notebook reader.</p>
</div>

![Screenshot%202024-10-26%20at%204.11.18%20PM.png](attachment:Screenshot%202024-10-26%20at%204.11.18%20PM.png)

In [ ]:
# Add in cells for 1.2 and complete all the work here.

<font size="4"> <b>Objective 1.2.1</b>
<br>There are 281 CFTR variants defined as "Pathogenic" in the Gnomad browser. 
    <br><font color="red">14 of these variants are present in the 1000 Genomes data.</font>
<br>Work is shown below.</font>

In [ ]:
# To start, CFTR variant data (.csv file) was downloaded from Gnomad browser
#
# R was used to isolate "Pathogenic" variants
#
# A screenshot of the code is included below
# The code is also included in our 'Question1.2' folder as 'Project6-PathogenicVariants-Real.R'
# The goal of this R code was to output a .txt file with a list of rsIDs for "Pathogenic" CFTR variants.
# The output 'rsIDs.txt' can also be found in the 'Question1.2' folder
#
# We have tried the suggested method of inserting pictures and apologize if you cannot see the screenshot
# All necessary information is still available in 'Project6-PathogenicVariants-Real.R'

![Screenshot%202024-11-08%20at%209.32.40%20PM.png](attachment:Screenshot%202024-11-08%20at%209.32.40%20PM.png)

In [ ]:
# Once the rsIDs.txt file was made, it was uploaded to Jupyter notebook and used in the code below.

In [1]:
# Import 1000 Genomes data for chromosome 7
# We double-checked in R that all "Pathogenic" CFTR variants are located on chromosome 7
!cp ~/biol6150/Data/1000Genomes/phase3.chr7.GRCh38.GT.crossmap.vcf.gz ~/biol6150/ProjectSubmissions/Group8/Project6/Question1.2
!cp ~/biol6150/Data/1000Genomes/phase3.chr7.GRCh38.GT.crossmap.vcf.gz.tbi ~/biol6150/ProjectSubmissions/Group8/Project6/Question1.2

In [1]:
# Isolate the "Pathogenic" CFTR variants from the 1000 Genomes data
!zgrep -F -f ~/biol6150/ProjectSubmissions/Group8/Project6/Question1.2/rsIDs.txt ~/biol6150/ProjectSubmissions/Group8/Project6/Question1.2/phase3.chr7.GRCh38.GT.crossmap.vcf.gz > ~/biol6150/ProjectSubmissions/Group8/Project6/Question1.2/pathogenicvariants.vcf
!wc -l ~/biol6150/ProjectSubmissions/Group8/Project6/Question1.2/pathogenicvariants.vcf

14 /home/hice1/acressy3/biol6150/ProjectSubmissions/Group8/Project6/Question1.2/pathogenicvariants.vcf


In [ ]:
# ^^The 14 lines in the 'pathogenicvariants.vcf' file represent the 14 CFTR "Pathogenic" variants present in the 1000 Genomes data.

<font size='4'> <b>Objective 1.2.2 </b><br>
<font color="red">BEB: 2
CEU: 1
CHB: 1
CHS: 1
GBR: 1
JPT: 1
LWK: 2
MSL: 1
MXL: 1
TSI: 2
YRI: 1</font>
<br> Work is shown below.</font>

In [1]:
# Download the population info file and move it into 'Question1.2 folder'
!wget https://ftp.1000genomes.ebi.ac.uk/vol1/ftp/release/20130502/integrated_call_samples_v3.20130502.ALL.panel
!mv ~/biol6150/ProjectSubmissions/Group8/Project6/integrated_call_samples_v3.20130502.ALL.panel ~/biol6150/ProjectSubmissions/Group8/Project6/Question1.2/integrated_call_samples_v3.20130502.ALL.panel

--2024-11-08 22:09:01--  https://ftp.1000genomes.ebi.ac.uk/vol1/ftp/release/20130502/integrated_call_samples_v3.20130502.ALL.panel
Resolving ftp.1000genomes.ebi.ac.uk (ftp.1000genomes.ebi.ac.uk)... 193.62.193.167
Connecting to ftp.1000genomes.ebi.ac.uk (ftp.1000genomes.ebi.ac.uk)|193.62.193.167|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 55156 (54K)
Saving to: ‘integrated_call_samples_v3.20130502.ALL.panel’

integrated_call_sam 100%[===================>]  53.86K   271KB/s    in 0.2s    

2024-11-08 22:09:02 (271 KB/s) - ‘integrated_call_samples_v3.20130502.ALL.panel’ saved [55156/55156]



In [1]:
# Extract the positions of the 14 CFTR variants
!zgrep -v "^#" ~/biol6150/ProjectSubmissions/Group8/Project6/Question1.2/pathogenicvariants.vcf | awk '{print $1 "\t" $2}' > ~/biol6150/ProjectSubmissions/Group8/Project6/Question1.2/positions.txt
# Find the IDs of individuals in 1000 Genomes data with at least one "Pathogenic" CFTR variant
!bcftools view -R ~/biol6150/ProjectSubmissions/Group8/Project6/Question1.2/positions.txt ~/biol6150/ProjectSubmissions/Group8/Project6/Question1.2/phase3.chr7.GRCh38.GT.crossmap.vcf.gz -o ~/biol6150/ProjectSubmissions/Group8/Project6/Question1.2/filtered_1000G_variants.vcf.gz -O z
!bcftools query -i 'GT~"1"' -f '%CHROM\t%POS\t[%SAMPLE\n]' ~/biol6150/ProjectSubmissions/Group8/Project6/Question1.2/filtered_1000G_variants.vcf.gz > ~/biol6150/ProjectSubmissions/Group8/Project6/Question1.2/individuals_with_variants.txt
!awk '{print $3}' ~/biol6150/ProjectSubmissions/Group8/Project6/Question1.2/individuals_with_variants.txt | sort | uniq > ~/biol6150/ProjectSubmissions/Group8/Project6/Question1.2/individual_ids.txt

In [1]:
#Associate individuals with their populations
!sort -k1,1 ~/biol6150/ProjectSubmissions/Group8/Project6/Question1.2/integrated_call_samples_v3.20130502.ALL.panel > ~/biol6150/ProjectSubmissions/Group8/Project6/Question1.2/sorted_panel.txt
!join -1 1 -2 1 ~/biol6150/ProjectSubmissions/Group8/Project6/Question1.2/individual_ids.txt ~/biol6150/ProjectSubmissions/Group8/Project6/Question1.2/sorted_panel.txt > ~/biol6150/ProjectSubmissions/Group8/Project6/Question1.2/individuals_with_population.txt
# Create a population summary
!awk '{print $2}' ~/biol6150/ProjectSubmissions/Group8/Project6/Question1.2/individuals_with_population.txt > ~/biol6150/ProjectSubmissions/Group8/Project6/Question1.2/populations.txt
!sort ~/biol6150/ProjectSubmissions/Group8/Project6/Question1.2/populations.txt | uniq -c > ~/biol6150/ProjectSubmissions/Group8/Project6/Question1.2/population_summary.txt
!awk '{print $2": " $1}' ~/biol6150/ProjectSubmissions/Group8/Project6/Question1.2/population_summary.txt

BEB: 2
CEU: 1
CHB: 1
CHS: 1
GBR: 1
JPT: 1
LWK: 2
MSL: 1
MXL: 1
TSI: 2
YRI: 1


# **2. Annotating variants**
### *Total Questions: 2*
### *Total Points: 35 = 10 + 10 + 15*


---

<div class="alert alert-block alert-warning">
    <h3>2.1 Find clinically important genes</h3>
    <p>Different genes perform different functions in our body. Is there a way we can find clincically important genes for humans? Search on web and find a resource that can tell you which genes are clinically important (there can be multiple answers)</p>
</div>

In [14]:
#What was the resource that you identified? Give a 1-3 line description and the web link.

ClinGen - https://clinicalgenome.org 

ClinGen is an NHGRI funded project that functions as a central resource for defining the clinical relevance of genes and variants to support research and precision medicine. It collaborates with ClinVar as its main source for variant data from clinicians, researchers, experts, and individual submitters. ClinGen works to evalutate relevance, establish standards for annotation and interpretation of variants, as well as develop precision medicine tools and create a more accessible user interface to promote the use of ClinVar and ClinGen. 

<div class="alert alert-block alert-warning">
    <h3>2.2 Varints in those genes</h3>
    <p><b>2.2.1</b> Identify a set of genes (3-8 genes; not a lot) that are clinically relevant for humans? These could be on one chromosome or multiple.</p>
    <p><b>2.2.2</b> Extract variants on the selected genes and annotate them. Depending on what genesyou selected, you might have to work on more than one VCF file.</p>
    <p><b>2.2.3</b> Pick any one of the genes and identify a transcript associated on that gene. Clearly write what transcript you have picked, and why?. How many variants have a HIGH or MODERATE variant effect consequence? (If your gene/transcript has 0 HIGH or MODERATE variants, please select another gene).</p>
</div>

## 2.2.1 Identified set of genes that are clinically relevant for humans 


- TP53 (chromosome 17): tumor supressor gene: mutation in TP53 can lead to wide range of cancer
- APC (chromosome 5): mutation in APC gene is associated with familial adenomatous polypopsis 
- RB1 (chromosome 13): gene associated with retinoblastoma 

## 2.2.2 Extracting variants on chr17,chr5,and chr13 in VCF files  

In [ ]:
# Getting chr13 file and moving it to folder Question2

In [2]:
!wget https://ftp.1000genomes.ebi.ac.uk/vol1/ftp/release/20130502/ALL.chr13.phase3_shapeit2_mvncall_integrated_v5b.20130502.genotypes.vcf.gz

--2024-11-08 10:46:27--  https://ftp.1000genomes.ebi.ac.uk/vol1/ftp/release/20130502/ALL.chr13.phase3_shapeit2_mvncall_integrated_v5b.20130502.genotypes.vcf.gz
Resolving ftp.1000genomes.ebi.ac.uk (ftp.1000genomes.ebi.ac.uk)... 193.62.193.167
Connecting to ftp.1000genomes.ebi.ac.uk (ftp.1000genomes.ebi.ac.uk)|193.62.193.167|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 533918831 (509M) [application/x-gzip]
Saving to: ‘ALL.chr13.phase3_shapeit2_mvncall_integrated_v5b.20130502.genotypes.vcf.gz’

ALL.chr13.phase3_sh 100%[===================>] 509.18M   114MB/s    in 5.5s    

2024-11-08 10:46:34 (93.0 MB/s) - ‘ALL.chr13.phase3_shapeit2_mvncall_integrated_v5b.20130502.genotypes.vcf.gz’ saved [533918831/533918831]



In [3]:
!mv ~/biol6150/ProjectSubmissions/Group8/Project6/ALL.chr13.phase3_shapeit2_mvncall_integrated_v5b.20130502.genotypes.vcf.gz ~/biol6150/ProjectSubmissions/Group8/Project6/Question2

In [15]:
# Indexing the chr 13 vcf file 
!bcftools index ~/biol6150/ProjectSubmissions/Group8/Project6/Question2/ALL.chr13.phase3_shapeit2_mvncall_integrated_v5b.20130502.genotypes.vcf.gz

In [17]:
# Filtering variants. The APC gene is located approximately at chr13: 48,303,744-48,599,436 forward strand.
!bcftools view -r 13:48303744-48599436 ~/biol6150/ProjectSubmissions/Group8/Project6/Question2/ALL.chr13.phase3_shapeit2_mvncall_integrated_v5b.20130502.genotypes.vcf.gz > ~/biol6150/ProjectSubmissions/Group8/Project6/Question2/RB1_variants.vcf

In [ ]:
# For chr5

In [6]:
!wget https://ftp.1000genomes.ebi.ac.uk/vol1/ftp/release/20130502/ALL.chr5.phase3_shapeit2_mvncall_integrated_v5b.20130502.genotypes.vcf.gz

--2024-11-08 14:36:10--  https://ftp.1000genomes.ebi.ac.uk/vol1/ftp/release/20130502/ALL.chr5.phase3_shapeit2_mvncall_integrated_v5b.20130502.genotypes.vcf.gz
Resolving ftp.1000genomes.ebi.ac.uk (ftp.1000genomes.ebi.ac.uk)... 193.62.193.167
Connecting to ftp.1000genomes.ebi.ac.uk (ftp.1000genomes.ebi.ac.uk)|193.62.193.167|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 946995712 (903M) [application/x-gzip]
Saving to: ‘ALL.chr5.phase3_shapeit2_mvncall_integrated_v5b.20130502.genotypes.vcf.gz’

ALL.chr5.phase3_sha 100%[===================>] 903.12M  96.1MB/s    in 11s     

2024-11-08 14:36:24 (79.9 MB/s) - ‘ALL.chr5.phase3_shapeit2_mvncall_integrated_v5b.20130502.genotypes.vcf.gz’ saved [946995712/946995712]



In [4]:
!mv ~/biol6150/ProjectSubmissions/Group8/Project6/ALL.chr5.phase3_shapeit2_mvncall_integrated_v5b.20130502.genotypes.vcf.gz ~/biol6150/ProjectSubmissions/Group8/Project6/Question2

In [21]:
# Indexing chr5 file and filtering variants at chr 21. 
# APC gene is located in Chromosome 5: 112,707,518-112,846,239 forward strand
!bcftools index ~/biol6150/ProjectSubmissions/Group8/Project6/Question2/ALL.chr5.phase3_shapeit2_mvncall_integrated_v5b.20130502.genotypes.vcf.gz
!bcftools view -r 5:112707518-112846239 ~/biol6150/ProjectSubmissions/Group8/Project6/Question2/ALL.chr5.phase3_shapeit2_mvncall_integrated_v5b.20130502.genotypes.vcf.gz > ~/biol6150/ProjectSubmissions/Group8/Project6/Question2/APC_variants.vcf

In [ ]:
# For chr17

In [6]:
!wget https://ftp.1000genomes.ebi.ac.uk/vol1/ftp/release/20130502/ALL.chr17.phase3_shapeit2_mvncall_integrated_v5b.20130502.genotypes.vcf.gz

--2024-11-08 10:50:37--  https://ftp.1000genomes.ebi.ac.uk/vol1/ftp/release/20130502/ALL.chr17.phase3_shapeit2_mvncall_integrated_v5b.20130502.genotypes.vcf.gz
Resolving ftp.1000genomes.ebi.ac.uk (ftp.1000genomes.ebi.ac.uk)... 193.62.193.167
Connecting to ftp.1000genomes.ebi.ac.uk (ftp.1000genomes.ebi.ac.uk)|193.62.193.167|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 415901551 (397M) [application/x-gzip]
Saving to: ‘ALL.chr17.phase3_shapeit2_mvncall_integrated_v5b.20130502.genotypes.vcf.gz’

ALL.chr17.phase3_sh  29%[====>               ] 116.82M  2.43MB/s    in 3m 10s  

2024-11-08 10:53:50 (631 KB/s) - Read error at byte 122495689/415901551 (Error decoding the received TLS packet.). Retrying.

--2024-11-08 10:53:51--  (try: 2)  https://ftp.1000genomes.ebi.ac.uk/vol1/ftp/release/20130502/ALL.chr17.phase3_shapeit2_mvncall_integrated_v5b.20130502.genotypes.vcf.gz
Connecting to ftp.1000genomes.ebi.ac.uk (ftp.1000genomes.ebi.ac.uk)|193.62.193.167|:443... connec

In [7]:
#moving the file to Question2 folder
!mv ~/biol6150/ProjectSubmissions/Group8/Project6/ALL.chr17.phase3_shapeit2_mvncall_integrated_v5b.20130502.genotypes.vcf.gz ~/biol6150/ProjectSubmissions/Group8/Project6/Question2

In [23]:
# indexing and extracting variants. 
# brca1 gene is located in Chromosome 17: 43,044,295-43,170,245 reverse strand.
!bcftools index ~/biol6150/ProjectSubmissions/Group8/Project6/Question2/ALL.chr17.phase3_shapeit2_mvncall_integrated_v5b.20130502.genotypes.vcf.gz
!bcftools view -r 17:43044295-43170245 ~/biol6150/ProjectSubmissions/Group8/Project6/Question2/ALL.chr17.phase3_shapeit2_mvncall_integrated_v5b.20130502.genotypes.vcf.gz > ~/biol6150/ProjectSubmissions/Group8/Project6/Question2/BRCA1_variants.vcf

In [26]:
# Annotating with snpEff 
!snpEff

SnpEff version SnpEff 5.1d (build 2022-04-19 15:49), by Pablo Cingolani
Usage: snpEff [command] [options] [files]

Run 'java -jar snpEff.jar command' for help on each specific command

Available commands: 
	[eff|ann]                    : Annotate variants / calculate effects (you can use either 'ann' or 'eff', they mean the same). Default: ann (no command or 'ann').
	build                        : Build a SnpEff database.
	buildNextProt                : Build a SnpEff for NextProt (using NextProt's XML files).
	cds                          : Compare CDS sequences calculated form a SnpEff database to the one in a FASTA file. Used for checking databases correctness.
	closest                      : Annotate the closest genomic region.
	count                        : Count how many intervals (from a BAM, BED or VCF file) overlap with each genomic interval.
	databases                    : Show currently available databases (from local config file).
	download                     : Download a

In [2]:
# Annotating RB1 Gene on chr13
!snpEff -Xmx16g ann -v GRCh38.105 ~/biol6150/ProjectSubmissions/Group8/Project6/Question2/ALL.chr13.phase3_shapeit2_mvncall_integrated_v5b.20130502.genotypes.vcf.gz > ~/biol6150/ProjectSubmissions/Group8/Project6/Question2/chr13_RB1_annotated.vcf

00:00:00 SnpEff version SnpEff 5.1d (build 2022-04-19 15:49), by Pablo Cingolani
00:00:00 Command: 'ann'
00:00:00 Reading configuration file 'snpEff.config'. Genome: 'GRCh38.105'
00:00:00 Reading config file: /storage/ice-shared/biol6150/ProjectSubmissions/Group8/Project6/snpEff.config
00:00:00 Reading config file: /storage/ice-shared/biol6150/SharedEnvironment/biol6150/share/snpeff-5.1-2/snpEff.config
00:00:00 done
00:00:00 Reading database for genome version 'GRCh38.105' from file '/storage/ice-shared/biol6150/SharedEnvironment/biol6150/share/snpeff-5.1-2/./data/GRCh38.105/snpEffectPredictor.bin' (this might take a while)
00:00:12 done
00:00:12 Loading Motifs and PWMs
00:00:12 Building interval forest
00:00:15 done.
00:00:15 Genome stats :
#-----------------------------------------------
# Genome name                : 'Homo_sapiens'
# Genome version             : 'GRCh38.105'
# Genome ID                  : 'GRCh38.105[0]'
# Has protein coding info    : true
# Has Tr. Support Level in

In [7]:
# Annotating APC Gene on chr5
!snpEff -Xmx16g ann -v GRCh38.105 ~/biol6150/ProjectSubmissions/Group8/Project6/Question2/ALL.chr5.phase3_shapeit2_mvncall_integrated_v5b.20130502.genotypes.vcf.gz > ~/biol6150/ProjectSubmissions/Group8/Project6/Question2/chr5_APC_annotated.vcf

00:00:00 SnpEff version SnpEff 5.1d (build 2022-04-19 15:49), by Pablo Cingolani
00:00:00 Command: 'ann'
00:00:00 Reading configuration file 'snpEff.config'. Genome: 'GRCh38.105'
00:00:00 Reading config file: /storage/ice-shared/biol6150/ProjectSubmissions/Group8/Project6/snpEff.config
00:00:00 Reading config file: /storage/ice-shared/biol6150/SharedEnvironment/biol6150/share/snpeff-5.1-2/snpEff.config
00:00:00 done
00:00:00 Reading database for genome version 'GRCh38.105' from file '/storage/ice-shared/biol6150/SharedEnvironment/biol6150/share/snpeff-5.1-2/./data/GRCh38.105/snpEffectPredictor.bin' (this might take a while)
00:00:12 done
00:00:12 Loading Motifs and PWMs
00:00:12 Building interval forest
00:00:15 done.
00:00:15 Genome stats :
#-----------------------------------------------
# Genome name                : 'Homo_sapiens'
# Genome version             : 'GRCh38.105'
# Genome ID                  : 'GRCh38.105[0]'
# Has protein coding info    : true
# Has Tr. Support Level in

In [1]:
# Annotating BRCA1 Gene on chr17
!snpEff -Xmx16g ann -v GRCh38.105 ~/biol6150/ProjectSubmissions/Group8/Project6/Question2/ALL.chr17.phase3_shapeit2_mvncall_integrated_v5b.20130502.genotypes.vcf.gz > ~/biol6150/ProjectSubmissions/Group8/Project6/Question2/chr17_BCRA1_annotated.vcf

00:00:00 SnpEff version SnpEff 5.1d (build 2022-04-19 15:49), by Pablo Cingolani
00:00:00 Command: 'ann'
00:00:00 Reading configuration file 'snpEff.config'. Genome: 'GRCh38.105'
00:00:00 Reading config file: /storage/ice-shared/biol6150/ProjectSubmissions/Group8/Project6/snpEff.config
00:00:00 Reading config file: /storage/ice-shared/biol6150/SharedEnvironment/biol6150/share/snpeff-5.1-2/snpEff.config
00:00:00 done
00:00:00 Reading database for genome version 'GRCh38.105' from file '/storage/ice-shared/biol6150/SharedEnvironment/biol6150/share/snpeff-5.1-2/./data/GRCh38.105/snpEffectPredictor.bin' (this might take a while)
00:00:12 done
00:00:12 Loading Motifs and PWMs
00:00:12 Building interval forest
00:00:15 done.
00:00:15 Genome stats :
#-----------------------------------------------
# Genome name                : 'Homo_sapiens'
# Genome version             : 'GRCh38.105'
# Genome ID                  : 'GRCh38.105[0]'
# Has protein coding info    : true
# Has Tr. Support Level in

## 2.2.3 Identify a transcript associated on that gene. How many variants have a HIGH or MODERATE variant effect consequence? 

In [8]:
# Identify a transcript associated 
!grep "BRCA1" ~/biol6150/ProjectSubmissions/Group8/Project6/Question2/chr17_BCRA1_annotated.vcf | grep -E "HIGH|MODERATE" | grep -oP "ENST[0-9]+" | sort | uniq -c

      1 ENST00000341165
    104 ENST00000352993
     78 ENST00000354071
      3 ENST00000356906
    104 ENST00000357654
     73 ENST00000412061
      3 ENST00000460115
    200 ENST00000461221
     84 ENST00000461574
     11 ENST00000461798
      3 ENST00000467245
    104 ENST00000468300
     78 ENST00000470026
    104 ENST00000471181
     14 ENST00000472490
     74 ENST00000473961
     78 ENST00000476777
     78 ENST00000477152
     97 ENST00000478531
     93 ENST00000484087
     87 ENST00000487825
      6 ENST00000489037
    104 ENST00000491747
     92 ENST00000492859
    104 ENST00000493795
     97 ENST00000493919
     78 ENST00000494123
     78 ENST00000497488
     13 ENST00000497954
      1 ENST00000542611
    104 ENST00000586385
      1 ENST00000586650
      1 ENST00000589047
      1 ENST00000589464
      1 ENST00000589872
      1 ENST00000590740
      1 ENST00000590996
    104 ENST00000591534
    104 ENST00000591849
      1 ENST00000592304
      3 ENST00000618469
     13 ENST0000

# There are total 2458 variants that have a high or moderate variant effect consequence. 

Summing up all the numbers in the left column:
104 + 78 + 3 + 104 + 73 + 3 + 200 + 84 + 11 + 3 + 104 + 78 + 104 + 14 + 74 + 78 + 78 + 97 + 93 + 87 + 6 + 104 + 92 + 104 + 97 + 78 + 78 + 13 + 1 + 104 + 1 + 1 + 1 + 1 + 1 + 1 + 104 + 104 + 1 + 3 + 13 + 80 + 1 + 1 + 1 + 1 + 87 + 97 + 6 + 78 + 1 + 3 + 1 = 2458